otx/algorithms/detection/adapters/mmdet/utils/config_utils.py

```diff
@@ -326,3 +327,58 @@ def cluster_anchors(recipe_config: Config, dataset: DatasetEntity):
     config_generator.widths, config_generator.heights = widths, heights
 
     recipe_config.model.bbox_head.anchor_generator = config_generator
+
+
+def patch_tiling(config, hparams, dataset=None):
+    """Update config for tiling.
+
+    Args:
+        config (dict): MPA config containing configuration settings.
+        hparams (DetectionConfig): DetectionConfig containing hyperparameters.
+        dataset (DatasetEntity, optional): A dataset entity. Defaults to None.
+
+    Returns:
+        dict: The updated configuration dictionary.
+    """
+    if hparams.tiling_parameters.enable_tiling:
+        logger.info("Tiling enabled")
+
+        if dataset and dataset.purpose != DatasetPurpose.INFERENCE and hparams.tiling_parameters.enable_adaptive_params:
+            adaptive_tile_params(hparams.tiling_parameters, dataset)
+            tiling_params = ConfigDict(
+                tile_size=int(hparams.tiling_parameters.tile_size),
+                overlap_ratio=float(hparams.tiling_parameters.tile_overlap),
+                max_per_img=int(hparams.tiling_parameters.tile_max_number),
+            )
+            config.update(
+                ConfigDict(
+                    data=ConfigDict(
+                        train=tiling_params,
+                        val=tiling_params,
+                        test=tiling_params,
+                    )
+                )
+            )
+
+        if dataset and dataset.purpose == DatasetPurpose.INFERENCE:
```

Comment:
Don't we have setting samples_per_gpu to 1 for test_dataloader? If we don't have that logic just resolve this. I think I saw that logic in the stage

Reply:
It is set to 2 (or 4, I can't remember) in our current code, although I don't know where it is set yet. 

Before I change this universally to 1, I'd like to confirm check the batch size should be set to 1? I think this is set to 1 before, but somehow set to > 1 and I thought this is intentional, no?

Reply:
Okay, then we should set validation batch size in the task level, I'll keep this in my mind for refactoring